src/ops/overflowing.rs

```diff
@@ -0,0 +1,101 @@
+use core::ops::{Add, Mul, Sub};
+#[cfg(has_i128)]
+use core::{i128, u128};
+use core::{i16, i32, i64, i8, isize};
+use core::{u16, u32, u64, u8, usize};
+
+macro_rules! overflowing_impl {
+    ($trait_name:ident, $method:ident, $t:ty) => {
+        impl $trait_name for $t {
+            #[inline]
+            fn $method(&self, v: &Self) -> (Self, bool) {
+                <$t>::$method(*self, *v)
+            }
+        }
+    };
+}
+
+/// If an overflow would have occurred then the wrapped value is returned.
```

Comment:
This line seems like it would better belong as the second sentence of the method docs. At the top we need a more general description, perhaps something simple like "Performs addition with a flag for overflow."

The same goes for OverflowingSub/Mul.